# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../curlpp/gcc43.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- curlpp-0.7.2/examples/example14.cpp.vanilla	2009-01-29 12:49:39.000000000 +0100
+++ curlpp-0.7.2/examples/example14.cpp	2009-01-29 12:49:55.000000000 +0100
@@ -21,6 +21,8 @@
  *    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include <stdlib.h>
+
 #include <iostream>
 
 #include <curlpp/cURLpp.hpp>
@@ -59,7 +61,8 @@
     requests.add(&request2);
     
     /* we start some action by calling perform right away */
-    while(!requests.perform(&nbLeft));
+    while(!requests.perform(&nbLeft))
+	;
     
     while(nbLeft) {
       struct timeval timeout;
@@ -94,7 +97,8 @@
       default:
 	/* one or more of curl's file descriptors say there's data to read
 	   or write */
-	while(!requests.perform(&nbLeft));
+	while(!requests.perform(&nbLeft))
+		;
 	break;
       }
     }
--- curlpp-0.7.2/examples/example11.cpp.vanilla	2009-01-29 12:48:12.000000000 +0100
+++ curlpp-0.7.2/examples/example11.cpp	2009-01-29 12:48:36.000000000 +0100
@@ -23,6 +23,7 @@
 
 #include <stdlib.h>
 #include <errno.h>
+#include <string.h>
 
 #include <curlpp/cURLpp.hpp>
 #include <curlpp/Easy.hpp>
--- curlpp-0.7.2/examples/example13.cpp.vanilla	2009-01-29 12:48:49.000000000 +0100
+++ curlpp-0.7.2/examples/example13.cpp	2009-01-29 12:49:28.000000000 +0100
@@ -21,6 +21,8 @@
  *    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include <stdlib.h>
+
 #include <iostream>
 
 #include <curlpp/cURLpp.hpp>
@@ -59,7 +61,8 @@
     requests.add(&request2);
     
     /* we start some action by calling perform right away */
-    while(!requests.perform(&nbLeft));
+    while(!requests.perform(&nbLeft))
+	;
     
     while(nbLeft) {
       struct timeval timeout;
@@ -92,7 +95,8 @@
       case 0:
       default:
 	/* timeout or readable/writable sockets */
-	while(!requests.perform(&nbLeft));
+	while(!requests.perform(&nbLeft))
+		;
 	break;
       }
     }
--- curlpp-0.7.2/examples/example10.cpp.vanilla	2009-01-29 12:47:52.000000000 +0100
+++ curlpp-0.7.2/examples/example10.cpp	2009-01-29 12:48:01.000000000 +0100
@@ -21,6 +21,7 @@
  *    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 
--- curlpp-0.7.2/examples/example21.cpp.vanilla	2009-01-29 12:50:36.000000000 +0100
+++ curlpp-0.7.2/examples/example21.cpp	2009-01-29 12:50:52.000000000 +0100
@@ -25,6 +25,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 #include <curlpp/cURLpp.hpp>
 #include <curlpp/Easy.hpp>
--- curlpp-0.7.2/examples/example06.cpp.vanilla	2009-01-29 12:47:08.000000000 +0100
+++ curlpp-0.7.2/examples/example06.cpp	2009-01-29 12:47:17.000000000 +0100
@@ -22,6 +22,7 @@
  */
 
 #include <stdlib.h>
+#include <string.h>
 
 #include <curlpp/cURLpp.hpp>
 #include <curlpp/Easy.hpp>
--- curlpp-0.7.2/examples/example02.cpp.vanilla	2009-01-29 12:46:33.000000000 +0100
+++ curlpp-0.7.2/examples/example02.cpp	2009-01-29 12:46:42.000000000 +0100
@@ -23,6 +23,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 #include <curlpp/cURLpp.hpp>
 #include <curlpp/Easy.hpp>
--- curlpp-0.7.2/examples/example05.cpp.vanilla	2009-01-29 12:46:54.000000000 +0100
+++ curlpp-0.7.2/examples/example05.cpp	2009-01-29 12:47:02.000000000 +0100
@@ -22,6 +22,7 @@
  */
 
 #include <stdlib.h>
+#include <string.h>
 
 #include <curlpp/cURLpp.hpp>
 #include <curlpp/Easy.hpp>
--- curlpp-0.7.2/curlpp/cURLpp.cpp.vanilla	2009-01-29 12:45:46.000000000 +0100
+++ curlpp-0.7.2/curlpp/cURLpp.cpp	2009-01-29 12:46:04.000000000 +0100
@@ -2,6 +2,8 @@
 #include "cURLpp.hpp"
 #include "Exception.hpp"
 
+#include <stdlib.h>
+
 void cURLpp::initialize( long flags )
 {
   CURLcode code;
--- curlpp-0.7.2/curlpp/CurlHandle.cpp.vanilla	2009-01-29 12:45:23.000000000 +0100
+++ curlpp-0.7.2/curlpp/CurlHandle.cpp	2009-01-29 12:45:31.000000000 +0100
@@ -21,6 +21,8 @@
  *    SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include <string.h>
+
 #include <string>
 #include <iostream>
 
--- curlpp-0.7.2/utilspp/singleton/LifetimeWithLongevity.inl.vanilla	2009-01-29 12:45:05.000000000 +0100
+++ curlpp-0.7.2/utilspp/singleton/LifetimeWithLongevity.inl	2009-01-29 12:45:10.000000000 +0100
@@ -1,3 +1,5 @@
+#include <algorithm>
+
 template< typename T, typename TDestroyer >
 void
 utilspp::setLongevity( T *obj, unsigned int longevity, TDestroyer d )
--- curlpp-0.7.2/utilspp/singleton/LifetimeLibrary.cpp.vanilla	2009-01-29 12:41:35.000000000 +0100
+++ curlpp-0.7.2/utilspp/singleton/LifetimeLibrary.cpp	2009-01-29 12:43:07.000000000 +0100
@@ -1,6 +1,9 @@
 #include "SingletonHolder.hpp"
 #include "LifetimeLibrary.hpp"
 
+#include <algorithm>
+
+
 utilspp::LifetimeLibraryImpl::LifetimeLibraryImpl()
    :
    mTrackerArray( NULL ),
